dask/dataframe/io/demo.py

```diff
@@ -8,15 +13,139 @@
+    """For a "category" or str column, list of possible values"""
+
+    low: int | None = None
+    """Start value for an int column. Optional if random=True, since ``randint`` doesn't accept
+    high and low."""
+
+    high: int | None = None
+    """For a "category" column, how many unique categories to generate"""
+
+    length: int | None = None
+    """For a str or "category" column, how many unique categories to generate"""
+
+    random: bool = False
+    """For an int column, whether to use ``randint``"""
+
+    method: str | None = None
+    """For an int column, method to use when generating the value, such as "poisson", "uniform", "binomial".
+    Default "poisson". Delegates to the same method of ``RandomState``"""
+
+    kwargs: dict[str, Any] = field(default_factory=dict)
+    """For an int column, any other kwargs to pass into the method"""
+
+
+@dataclass
+class IndexSpec:
+    """Properties of the dataframe index
 
+    Notes
+    -----
+    This API is still experimental, and will likely change in the future"""
 
-def make_float(n, rstate):
+    dtype: str | type = int
+    """Index dtype. Currently only supports integer and datetime types"""
+
+    start: str | None = None
+    """First value of the index, only required for a DatetimeIndex"""
+
+    freq: int | str = 1
+    """Step for a RangeIndex, frequency for a DatetimeIndex ("1H", "1D", etc.)"""
+
+    partition_freq: str | None = None
+    """Partition frequency, required for a DatetimeIndex ("1D", "1M", etc.)"""
+
+
+@dataclass
+class DatasetSpec:
+    """Defines a dataset with random data, such as which columns and data types to generate
+
+    Notes
+    -----
+    This API is still experimental, and will likely change in the future"""
+
+    npartitions: int = 1
+    """How many partitions generate in the dataframe. If the dataframe has a DatetimeIndex, specify
+    its ``partition_freq`` instead"""
+
+    nrecords: int = 1000
+    """Total number of records to generate"""
+
+    index_spec: IndexSpec = field(default_factory=IndexSpec)
+    """Properties of the index"""
+
+    column_specs: list[ColumnSpec] = field(default_factory=list)
+    """List of column definitions"""
+
+
+def make_float(n, rstate, random=False, dtype=None, **kwargs):
+    if random:
+        data = rstate.random(size=n, **kwargs)
+        if dtype:
+            data = data.astype(dtype)
+        return data
     return rstate.rand(n) * 2 - 1
 
 
-def make_int(n, rstate, lam=1000):
-    return rstate.poisson(lam, size=n)
+def make_int(
+    n: int,
+    rstate: Any,
+    random: bool = False,
+    dtype: str | type = int,
+    method: str | Callable = "poisson",
+    **kwargs,
+):
+    if random:
+        data = rstate.randint(size=n, **kwargs)
+    else:
+        if isinstance(method, str):
+            # "poisson", "binomial", etc.
+            handler_args, handler_kwargs = default_int_args.get(method, ((), {}))
+            handler_kwargs = copy.copy(handler_kwargs)
```

Comment:
Is there a difference between copy(...) and handler_kwargs.copy()?

Reply:
Nope, same thing.